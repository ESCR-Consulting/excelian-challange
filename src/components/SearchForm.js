import React from 'react'
import { Field, reduxForm } from 'redux-form'
import Paper from 'material-ui/Paper';
import { withStyles } from 'material-ui/styles';
import SearchInput from './SearchInput'
import Search from 'material-ui-icons/Search';

const styles = theme => ({
    container: {
        display: 'flex',
        flexWrap: 'wrap',
    },
});

let SearchForm = ({ handleSubmit, classes }) =>
    <Paper>
        <form onSubmit={handleSubmit}>
            <div className={classes.container}>
                <label htmlFor="search"><Search/></label>
                <Field name="search" component={SearchInput} type="text" />
            </div>
            <button type="submit" style={{display: 'none'}}>Submit</button>
        </form>
    </Paper>


SearchForm = reduxForm({
    form: 'search'
})(SearchForm)

export default withStyles(styles)(SearchForm)
