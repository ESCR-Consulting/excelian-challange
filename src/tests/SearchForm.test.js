import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import SearchForm from '../components/SearchForm';

const classes = {
    container: '[object Object]',
    form: '[object Object]',
}
configure({ adapter: new Adapter() });

describe('SearchForm', () => {
    it('should render correctly', () => {
        const output = shallow(
            <SearchForm classes={classes}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
