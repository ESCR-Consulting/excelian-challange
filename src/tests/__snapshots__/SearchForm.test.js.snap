// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly 1`] = `
<SearchForm
  classes={
    Object {
      "container": "SearchForm-container-1 [object Object]",
      "form": "SearchForm-form-2 [object Object]",
    }
  }
/>
`;
